pkg/discovery/mock/mock.go

```diff
@@ -5,34 +5,36 @@
 package mock
 
 import (
+	"fmt"
 	"sync"
 
+	"github.com/ethersphere/bee/pkg/discovery"
 	"github.com/ethersphere/bee/pkg/swarm"
+
 	ma "github.com/multiformats/go-multiaddr"
 )
 
 type Discovery struct {
 	mtx     sync.Mutex
 	ctr     int //how many ops
-	records map[string]broadcastRecord
-}
-
-type broadcastRecord struct {
-	overlay   swarm.Address
-	multiaddr ma.Multiaddr
+	records map[string]discovery.BroadcastRecord
 }
 
 func NewDiscovery() *Discovery {
 	return &Discovery{
-		records: make(map[string]broadcastRecord),
+		records: make(map[string]discovery.BroadcastRecord),
 	}
 }
 
-func (d *Discovery) BroadcastPeer(addressee swarm.Address, overlay swarm.Address, addr ma.Multiaddr) error {
-	d.mtx.Lock()
-	defer d.mtx.Unlock()
-	d.ctr++
-	d.records[addressee.String()] = broadcastRecord{overlay: overlay, multiaddr: addr}
+func (d *Discovery) BroadcastPeers(addressee swarm.Address, peers ...discovery.BroadcastRecord) error {
+	for _, peer := range peers {
+		d.mtx.Lock()
```

Comment:
Hmmm, it depends what we consider better? Since this is mock, I though it would be ok to get more granularity on the account of performance, since it is not really overly optimized anyway. What do you think?

Reply:
That is fine in case of the mock.